core/src/main/scala/com/evolutiongaming/kafka/flow/RebalanceListener.scala

```diff
@@ -0,0 +1,62 @@
+package com.evolutiongaming.kafka.flow
+
+import cats.data.NonEmptySet
+import cats.syntax.all._
+import com.evolutiongaming.catshelper.{BracketThrowable, LogOf}
+import com.evolutiongaming.kafka.flow.ConsumerFlow.log
+import com.evolutiongaming.kafka.flow.kafka.Consumer
+import com.evolutiongaming.skafka.{Partition, Topic, TopicPartition}
+import com.evolutiongaming.skafka.consumer.{RebalanceListener => SRebalanceListener}
+
+object RebalanceListener {
+
+  def apply[F[_]: BracketThrowable: LogOf](
+    consumer: Consumer[F],
+    flows: Map[Topic, TopicFlow[F]]
+  ): SRebalanceListener[F] = new SRebalanceListener[F] {
+
+    def onPartitionsAssigned(topicPartitions: NonEmptySet[TopicPartition]) =
+      foreachTopic(topicPartitions) traverse_ { case (topic, flow, partitions) =>
+        for {
+          log <- log[F]
+          _ <- log.prefixed(topic).info(s"$partitions assigned")
+          partitions <- partitions.toNonEmptyList traverse { partition =>
+            consumer.position(TopicPartition(topic, partition)) map (partition -> _)
+          }
+          _ <- log.prefixed(topic).info(s"committed offsets: $partitions")
+          _ <- flow.add(partitions.toNes)
+        } yield ()
+      }
+
+    def onPartitionsRevoked(topicPartitions: NonEmptySet[TopicPartition]) =
+      foreachTopic(topicPartitions) traverse_ { case (topic, flow, partitions) =>
+        for {
+          log <- log[F]
+          _ <- log.prefixed(topic).info(s"$partitions revoked, removing from topic flow")
+          _ <- flow.remove(partitions)
+        } yield ()
+      }
+
+    def onPartitionsLost(topicPartitions: NonEmptySet[TopicPartition]) =
+      foreachTopic(topicPartitions) traverse_ { case (topic, flow, partitions) =>
+        for {
+          log <- log[F]
+          _ <- log.prefixed(topic).info(s"$partitions lost, removing from topic flow")
+          _ <- flow.remove(partitions)
+        } yield ()
+      }
+
+    private def foreachTopic[A](
+      topicPartitions: NonEmptySet[TopicPartition]
+    ): List[(Topic, TopicFlow[F], NonEmptySet[Partition])] =
```

Comment:
- The method name `foreachTopic` might not be ideal anymore, because the method does not have a function argument now.
- You dot need `private` modifier anymore as this is an anonymous implementation of `SRebalanceListener` now, so the method is not visible anymore.

Reply:
Renamed